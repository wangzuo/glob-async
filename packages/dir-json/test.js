const path = require('path');
const dirJSON = require('./');

function fn(x) {
  if (x.match(/node_modules/)) return false;
  if (path.basename(x)[0] === '.') return false;
  if (x.match(/__snapshots__/)) return false;
  if (x.match(/coverage/)) return false;
}

test('directory', async () => {
  expect(await dirJSON(path.join(__dirname, '../../'), fn)).toMatchSnapshot();
});

test('file', async () => {
  expect(
    await dirJSON(path.join(__dirname, '../../README.md'), fn)
  ).toMatchSnapshot();
});

test('overwrite directory', async () => {
  expect(
    await dirJSON(path.join(__dirname, '../../'), function(x) {
      if (x === '/packages/dir-json') return { children: [] };
      return fn(x);
    })
  ).toMatchSnapshot();
});
