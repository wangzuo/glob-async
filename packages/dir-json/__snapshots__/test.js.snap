// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directory 1`] = `
Object {
  "children": Array [
    Object {
      "id": "/README.md",
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "/packages/dir-json/index.js",
              "type": "file",
            },
            Object {
              "id": "/packages/dir-json/package.json",
              "type": "file",
            },
            Object {
              "id": "/packages/dir-json/test.js",
              "type": "file",
            },
            Object {
              "id": "/packages/dir-json/yarn.lock",
              "type": "file",
            },
          ],
          "id": "/packages/dir-json",
          "type": "directory",
        },
        Object {
          "children": Array [
            Object {
              "id": "/packages/glob-async/LICENSE",
              "type": "file",
            },
            Object {
              "id": "/packages/glob-async/README.md",
              "type": "file",
            },
            Object {
              "id": "/packages/glob-async/index.js",
              "type": "file",
            },
            Object {
              "id": "/packages/glob-async/package.json",
              "type": "file",
            },
            Object {
              "id": "/packages/glob-async/test.js",
              "type": "file",
            },
            Object {
              "id": "/packages/glob-async/yarn.lock",
              "type": "file",
            },
          ],
          "id": "/packages/glob-async",
          "type": "directory",
        },
        Object {
          "children": Array [
            Object {
              "id": "/packages/head-async/file.txt",
              "type": "file",
            },
            Object {
              "id": "/packages/head-async/index.js",
              "type": "file",
            },
            Object {
              "id": "/packages/head-async/package.json",
              "type": "file",
            },
            Object {
              "id": "/packages/head-async/test.js",
              "type": "file",
            },
            Object {
              "id": "/packages/head-async/yarn.lock",
              "type": "file",
            },
          ],
          "id": "/packages/head-async",
          "type": "directory",
        },
      ],
      "id": "/packages",
      "type": "directory",
    },
  ],
  "id": "/",
  "type": "directory",
}
`;

exports[`file 1`] = `
Object {
  "id": "/",
  "type": "file",
}
`;

exports[`overwrite directory 1`] = `
Object {
  "children": Array [
    Object {
      "id": "/README.md",
      "type": "file",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "/packages/dir-json/index.js",
              "type": "file",
            },
            Object {
              "id": "/packages/dir-json/package.json",
              "type": "file",
            },
            Object {
              "id": "/packages/dir-json/test.js",
              "type": "file",
            },
            Object {
              "id": "/packages/dir-json/yarn.lock",
              "type": "file",
            },
          ],
          "id": "/packages/dir-json",
          "type": "directory",
        },
      ],
      "id": "/packages",
      "type": "directory",
    },
  ],
  "id": "/",
  "type": "directory",
}
`;
